File: code_examples/3-Structured_Problem_Solving/2-Constants/Java/Main.java
/*
 * This program shows declaring constants
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

final class Main {
    private Main() {
        // Prevent instantiation
        // Optional: throw an exception e.g. AssertionError
        // if this ever *is* called
        throw new IllegalStateException("Cannot be instantiated");
    }

    /**
     * Constant number of lives.
     */
    private static final int ROOM = 212;

    /**
     * Constant for HST.
     */
    private static final double HST = 0.13;

    /**
     * Constant for COUNTRY.
     */
    private static final String COUNTRY = "Canada";

    /**
     * Main entry point into program.
     * @param args nothing passed in
     */
    public static void main(final String[] args) {
        // output
        System.out.println("Room: " + ROOM);
        System.out.println(HST + "%");
        System.out.println(COUNTRY);

        System.out.println("\nDone.");
    }
}

File: code_examples/3-Structured_Problem_Solving/2-Constants/Python3/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module shows declaring constants
"""

import constants


def main() -> None:
    """The main() function shows declaring constants, returns None."""
    print("Room: " + str(constants.ROOM_NUMBER))
    print(str(constants.HST) + "%")
    print(constants.COUNTRY)

    print("\nDone.")


if __name__ == "__main__":
    main()
